src/ui/groups_widgets.cc

```diff
@@ -772,6 +773,57 @@ void DictGroupsWidget::addAutoGroups()
   }
 }
 
+
+void DictGroupsWidget::addAutoGroupsByFolders()
+{
+  if ( activeDicts->empty() ) {
+    return;
+  }
+
+  if ( QMessageBox::information( this,
+                                 tr( "Confirmation" ),
+                                 tr( "Are you sure you want to generate a set of groups "
+                                     "based on containing folders?" ),
+                                 QMessageBox::Yes,
+                                 QMessageBox::Cancel )
+       != QMessageBox::Yes ) {
+    return;
+  }
+
+  QMultiMap< QString, sptr< Dictionary::Class > > dictMap;
+
+  for ( const auto & dict : *activeDicts ) {
+    QString dictFolder = dict->getContainingFolder();
+
+    if ( dictFolder.isEmpty() ) {
+      continue;
+    }
+
+    QDir dir = dictFolder;
+    if ( dir.cdUp() ) {
```

Comment:
dicFolder is a QString , will this cause trouble  QDir=QString

Reply:
No, QDir has an overridden `operator=`, so it will act like `auto dir = QDir(dictFolder)`